### ISSUE TITLE
Email sent every day during 7 days

### ISSUE THREAD

**Issue by dmekersa** [strangerstudios/pmpro-recurring-emails#7]
After checking to my email logs, the email is not send 7 days before the payment, but every day during 7 days. It impacts the cancellation decision in the mind of the customer.

**Comment by positivejam** [comment]
I also have this problem, and have been using a temporary fix. I've posted it below, but please use at your own risk; it differs from the version I'm using on my site, so I've never tested this _specific_ code. It differs from my actual code in two ways:

1. Some small changes that are specific to my site.
2. Has extra checks to make doubly sure not to email the user more than once. I now think they're probably unnecessary, I just haven't gotten around to testing that hypothesis.

Anyway, in case it's useful to anyone else, here's the code. **NOTE:** this code assumes your setup sends out reminders **_7 days_** before renewal. You'll need to adjust it if your site behaves differently.
```php
// PMPro should determine whether a member has already been notified, but it appears
// the sql query it uses has a bug. Perform this extra check to prevent users who've 
// already received a notice from getting another. If this is fixed in future versions of
// PMPro Recurring Emails Add On this filter can be removed.
function temp_filter_send_reminder_to_user ( $send_mail, $user, $lastorder ) {

	if ( ! $send_mail ) {
		// if we *weren't* going to email anyway, no logic below changes that
		return $send_mail;
	}

	$level = $user->data->membership_level;

	// NOTE: assumes your email goes out 7 days before renewal. Adjust as necessary.
	$was_emailed = get_user_meta( $user->ID, 'pmpro_recurring_notice_7', true );
	
	// If they've already been emailed about the upcoming payment, don't email again
	if ( $was_emailed ) {
		$send_mail = false;
	}
	
	return $send_mail;
}
add_filter( 'pmprorm_send_reminder_to_user', 'temp_filter_send_reminder_to_user', 10, 3 );
```

**Comment by eighty20results** [comment]
Believe this is resolved in pull request #9

**Comment by ideadude** [comment]
I'm going to close this. The latest update should fix this, but let us know (open a new issue) if you are still having problems after that.
